### DISCUSSION TITLE
How to make it remember the past conversation?

### DISCUSSION THREAD

**Discussion by reluctantslave** [safevideo/autollm#137] [Q&A] [answered]
Hey thanks for making this, it makes the process of building LLM-based application so much easier for absolute beginner like me. I've been playing with it for a while, and noticing it doesn't have capability to remember the past history conversation, on top of that it has short length response.

How to make it to remember the past history conversation, and make the response much longer and useful? I noticed in the config.yaml file that you can specify more than two tasks, can we combine it so that when a request has been made it will able to execute more than 2 tasks?

**Comment by talhasarit** [comment, marked as answer]
Hello @capsaicinoids,

We're thrilled to hear that our `autollm` package has been helpful to you as you explore building LLM-based applications! Your feedback is invaluable to us, especially coming from users who are just beginning their journey into this exciting field.

Regarding your queries:

1. **Conversation Memory:**
   At the moment, `autollm` is designed to handle stateless interactions, which means it doesn't inherently remember past interactions or the history of the conversation. This stateless design is intentional to simplify the model's operations for single-query scenarios, which are common in many applications. If you require a conversational memory feature, this would indeed be a substantial enhancement. You're welcome to open a feature request detailing your requirements, and we can consider it for future development. Your detailed description of the desired functionality will be very important for us to understand your needs better.

2. **Lengthier and More Useful Responses:**
   For generating longer and more informative responses, you can adjust the `max_tokens` parameter in the `config.yaml` file. This parameter dictates the maximum length of the output. To be able to use this feature, please update `autollm` to latest version by using `pip install -U autollm`.

  Please refer to our updated [`config.example.yaml`](https://github.com/safevideo/autollm/blob/main/examples/configs/config.example.yaml) file for a detailed guide on what parameters you can set:

   ```yaml
   # config.yaml

   llm_params:
     max_tokens: 1024  # Increase this value for longer outputs
   ```
3. **Executing Multiple Tasks:**
   The feature to execute multiple tasks is supported through our `AutoFastAPI` class, which provides the capability to handle different tasks by mapping them to specific query engines. This class allows you to configure multiple tasks in your `config.yaml` file, and the API will be able to execute the corresponding task based on the request it receives.

   Here's a snippet from our documentation showing how you can set up your `AutoFastAPI` instance to handle multiple tasks:

   ```python
   # Import necessary classes and functions
   from autollm.auto.fastapi_app import AutoFastAPI
   import uvicorn

   # Start the server using from_config class method
   app = AutoFastAPI.from_config("config.yaml", ".env")

   # Start the server
   uvicorn.run(app, host="0.0.0.0", port=8000)
   ```

   In the `config.yaml` file, you can specify the different tasks and their corresponding query engines. When a request is made to the `/query` endpoint with the specified task, the `AutoFastAPI` will use the appropriate query engine to execute the task and return the response.

   If you need further assistance on configuring this or have more specific requirements, please let us know, and we'd be happy to help!
